test/WebSites/ModelBindingWebSite/TestBindingSourceModelBinder.cs

```diff
@@ -8,10 +8,16 @@
 
 namespace ModelBindingWebSite
 {
-    public class TestMetadataAwareBinder : MetadataAwareBinder<FromTestAttribute>
+    public class TestBindingSourceModelBinder : BindingSourceModelBinder
     {
-        protected override Task<bool> BindAsync(ModelBindingContext bindingContext, FromTestAttribute metadata)
+        public TestBindingSourceModelBinder()
+            : base(FromTestAttribute.TestBindingSource)
         {
+        }
+
+        protected override Task BindModelCoreAsync(ModelBindingContext bindingContext)
+        {
+            var metadata = (FromTestAttribute)bindingContext.ModelMetadata.BinderMetadata;
```

Comment:
Yeah.. that was one nice thing about the old design. However the only place where a binder-metadata carried extra data was this test sooooooooooo